sys/benchmark/benchmark.c

```diff
@@ -26,7 +26,7 @@
 void benchmark_print_time(uint32_t time, unsigned long runs, const char *name)
 {
     uint32_t full = (time / runs);
-    uint32_t div  = (time - (full * runs)) / (runs / 1000);
+    uint32_t div  = (time - (full * runs)) * 1000 / runs;
```

Comment:
this might now overflow for values > (2**32)*1000?

Reply:
time is uint32 -> full * runs is  uint32 -> time - (full * runs) is uint32 (without the full seconds) -> 0..1 000 000 * 1000 ->  log2(10^9)  ~> 30 Bit  
pass

Reply:
This is only the remainder (`time % runs`) so this should not be an issue

Reply:
if (time > ~2^22 && runs > time) there might be an overflow  (

runs > time -> full = 0 -> time - ( full * runs) == time -> * 1000 ->   > 2^32

Reply:
If runs > time the result of `time - (full * runs)` will be negative

Reply:
no since full will be 0  (time / runs)